ibis/mapd/client.py

```diff
@@ -167,6 +185,44 @@ def __exit__(self, exc_type, exc_value, traceback):
         pass
 
 
+class MapDGeoCursor(MapDDefaultCursor):
+    """Cursor to allow the MapD client to reuse machinery in ibis/client.py
+    """
+
+    def to_df(self):
+        if not isinstance(self.cursor, Cursor):
+            if self.cursor is None:
+                return geopandas.GeoDataFrameDataFrame([])
+            return self.cursor
+
+        col_names = [c.name for c in self.cursor.description]
+        result = pd.DataFrame(self.cursor.fetchall(), columns=col_names)
+
+        # get geo types from pymapd
+        geotypes = (
+            pymapd_dtype.POINT,
+            pymapd_dtype.LINESTRING,
+            pymapd_dtype.POLYGON,
+            pymapd_dtype.MULTIPOLYGON,
+            pymapd_dtype.GEOMETRY,
+            pymapd_dtype.GEOGRAPHY,
+        )
+
+        geo_column = None
+
+        for d in self.cursor.description:
+            field_name = d.name
+            if d.type_code in geotypes:
+                # use the first geo column found as default geometry
```

Comment:
Can you explain this a comment a bit? Is it possible to have multiple geometries in a GeoDataFrame?